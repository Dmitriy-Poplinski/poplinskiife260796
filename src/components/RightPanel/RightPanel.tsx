import React from 'react';

import classes from './RightPanel.module.css';

interface RightPanelProps {
    image: string
}

export function RightPanel({ image }: RightPanelProps) {
    let color;

    try {
        color = `url(${require(`../../assets/images/${image}.jpg`)})`; 
    } catch (e) {
        color = 'maroon'
    };

    return (
        <div 
            className={classes.right_panel}
            style={{
                background: color
            }}
            data-testid="right_panel"
        >
            Right Panel
        </div>
    );
}
