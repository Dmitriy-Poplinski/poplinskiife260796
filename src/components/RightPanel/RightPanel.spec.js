import { render, screen } from '@testing-library/react';

import { RightPanel } from './RightPanel';

describe('RightPanel:', () => {

    let data; 

    test('RightPanel gets incorrect string', () => {
        data = 'testing_image';

        render(<RightPanel image={data} />);

        expect(screen.getByTestId('right_panel')).toHaveStyle('background: maroon');
        expect(screen.getByTestId('right_panel')).toMatchSnapshot();
    });

    test('RightPanel gets correct string', () => {
        data = 'right_panel_cover';

        render(<RightPanel image={data} />);

        expect(screen.getByTestId('right_panel')).toHaveStyle(`background: url("right_panel_cover.jpg")`);
        expect(screen.getByTestId('right_panel')).toMatchSnapshot();
    });
});
