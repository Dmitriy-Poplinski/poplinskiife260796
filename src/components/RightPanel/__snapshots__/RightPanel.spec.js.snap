// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RightPanel: RightPanel gets correct string 1`] = `
<div
  class="right_panel"
  data-testid="right_panel"
  style="background: url(right_panel_cover.jpg);"
>
  Right Panel
</div>
`;

exports[`RightPanel: RightPanel gets incorrect string 1`] = `
<div
  class="right_panel"
  data-testid="right_panel"
  style="background: maroon;"
>
  Right Panel
</div>
`;
